src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,46 @@
+package duke.tasks;
+
+
+public abstract class Task {
+    protected boolean isDone;
+    protected String taskName;
+
+    /**
+     *  Task constructor.
+     *
+     *  @param taskName represents the name of the task.
+     */
+    public Task(String taskName) {
+        this.taskName = taskName;
+        this.isDone = false;
+    }
+
+    //when task is done change the isDone to true
+    public void taskDone() {
```

Comment:
Perhaps a better name for the setter method of the `isDone` boolean variable could be `setDone`